packages/block-editor/src/store/reducer.js

```diff
@@ -2131,6 +2140,524 @@ const combinedReducers = combineReducers( {
+	}
+) {
+	if ( isZoomedOut || isNavMode ) {
+		// If the root block is the section root set its editing mode to contentOnly.
+		if ( clientId === sectionRootClientId ) {
+			return 'contentOnly';
+		}
+
+		// There are no sections, so everything else is disabled.
+		if ( ! sectionClientIds?.length ) {
+			return 'disabled';
+		}
+
+		if ( sectionClientIds.includes( clientId ) ) {
+			return 'contentOnly';
+		}
+
+		// If zoomed out, all blocks that aren't sections or the section root are
+		// disabled.
+		// If the tree root is not in a section, set its editing mode to disabled.
+		if (
+			isZoomedOut ||
+			! findParentInClientIdsList( state, clientId, sectionClientIds )
+		) {
+			return 'disabled';
+		}
+
+		// Next handle content blocks, which are either blocks with role: content
+		// or blocks that are inside synced patterns with bindings.
+		const block = state.blocks.tree.get( clientId );
+
+		// Handle synced pattern content so the inner blocks of a synced pattern are
+		// properly disabled.
+		if ( syncedPatternClientIds.length ) {
+			const parentPatternClientId = findParentInClientIdsList(
+				state,
+				clientId,
+				syncedPatternClientIds
+			);
+
+			if ( parentPatternClientId ) {
+				// This is a pattern nested in another pattern, it should be disabled.
+				if (
+					findParentInClientIdsList(
+						state,
+						parentPatternClientId,
+						syncedPatternClientIds
+					)
+				) {
+					return 'disabled';
+				}
+
+				if ( hasBindings( block ) ) {
+					return 'contentOnly';
+				}
+
+				// Synced pattern content without a binding isn't editable
+				// from the instance, the user has to edit the pattern source,
+				// so return 'disabled'.
+				return 'disabled';
+			}
+		}
+
+		if ( block?.name && isContentBlock( block.name ) ) {
+			return 'contentOnly';
+		}
+
+		return 'disabled';
+	}
+
+	if ( syncedPatternClientIds.length ) {
+		// Synced pattern blocks (core/block).
+		if ( syncedPatternClientIds.includes( clientId ) ) {
+			// This is a pattern nested in another pattern, it should be disabled.
+			if (
+				findParentInClientIdsList(
+					state,
+					clientId,
+					syncedPatternClientIds
+				)
+			) {
+				return 'disabled';
+			}
+
+			return 'contentOnly';
+		}
+
+		// Inner blocks of synced patterns.
+		const parentPatternClientId = findParentInClientIdsList(
+			state,
+			clientId,
+			syncedPatternClientIds
+		);
+		if ( parentPatternClientId ) {
+			// This is a pattern nested in another pattern, it should be disabled.
+			if (
+				findParentInClientIdsList(
+					state,
+					parentPatternClientId,
+					syncedPatternClientIds
+				)
+			) {
+				return 'disabled';
+			}
+
+			const block = state.blocks.tree.get( clientId );
+			if ( hasBindings( block ) ) {
+				return 'contentOnly';
+			}
+
+			// Synced pattern content without a binding isn't editable
+			// from the instance, the user has to edit the pattern source,
+			// so return 'disabled'.
+			return 'disabled';
+		}
+	}
+}
```

Comment:
Am I reading this wrong or we don't treat content locking parents?

Reply:
Oh answered in https://github.com/WordPress/gutenberg/pull/67026/files?diff=unified&w=1#r1849986579 - we're specifically NOT dealing with content locking. At least for now. I read this as a sweeping change. It's just one step.